satpy/readers/eum_l2_bufr.py

```diff
@@ -84,54 +94,95 @@ class SeviriL2BufrFileHandler(BaseFileHandler):
-            sc_id = self.get_attribute('satelliteIdentifier')
-            self.mpef_header = {}
-            self.mpef_header['NominalTime'] = buf_start_time
-            self.mpef_header['SpacecraftName'] = data_center_dict[sc_id]['name']
-            self.mpef_header['RectificationLongitude'] = data_center_dict[sc_id]['ssp']
+            # get all attributes in one call
+            attr = self.get_attributes(['typicalDate', 'typicalTime'])
+            timeStr = attr['typicalDate']+attr['typicalTime']
+
+            darr = self.get_array('satelliteIdentifier')
+            arr = da.where(darr != BUFR_FILL_VALUE, darr, darr).compute()
+            if (arr.ndim > 0):
+                sc_id = int(arr[0])
+            else:
+                sc_id = int(arr)
+
+            self.bufr_header = {}
+            self.bufr_header['NominalTime'] = datetime.strptime(timeStr, "%Y%m%d%H%M%S")
+            self.bufr_header['SpacecraftName'] = data_center_dict[sc_id]['name']
+            self.bufr_header['RectificationLongitude'] = data_center_dict[sc_id]['ssp']
 
         if rectification_longitude != 'default':
-            self.mpef_header['RectificationLongitude'] = f'E{int(rectification_longitude * 10):04d}'
-
-        self.with_adef = with_area_definition
-        if self.with_adef and filetype_info['file_type'] == 'seviri_l2_bufr_amv':
-            logging.warning("AMV BUFR data cannot be loaded with an area definition. Setting self.with_def = False.")
+            self.bufr_header['RectificationLongitude'] = f'E{int(rectification_longitude * 10):04d}'
+
+        self.filetype = filetype_info['file_type']
+        self.seg_size = seg_size_dict[self.filetype]
+        if self.seg_size:
+            # make this keyword not usable for non-grided products
+            self.with_adef = with_area_definition
+
+            darr = self.get_array('segmentSizeAtNadirInXDirection')
+            # This key may not exists in some files
+            # This is specifically a "dirty" fox for the unit tests because the template used doesn't
+            # contain that key so initializing the object would fail without thic check
+            # but 'real' EUM BUFR files shall all have it
+            if darr is None:
+                self.resolution = None
+            else:
+                arr = da.where(darr != BUFR_FILL_VALUE, darr, darr).compute()
+                self.resolution = arr[0]
+
+            # Note: There a difference between resolution and seg_size as from the seg_size_dict
+            # in the bufr files segmentSizeAtNadirInXDirection is encoded in meters so it's indeed
+            # the physical size of the segment while seg_size is the number of pixel lines / cols
+            # but because FCI pixel size is 1km they look very similar, modulo 1000
+        else:
             self.with_adef = False
-
-        self.seg_size = seg_size_dict[filetype_info['file_type']]
+            self.resolution = None
```

Comment:
Well, the key returns inconsistent values anyway. For instance for seviri AMV it's either 32000 or 72000 but I think it's supposed to be None for AMV, right?

Reply:
Hm, actually I think we need to have the resolution in the yaml-file. There is way to do it in the `.py` file but hardcoding it in the dictionary is not enough (this is because the resolution information is needed for the internal satpy processing). I'd say we keep the information in the yaml-file for now and can look into an alternative solution at a later stage.

Reply:
regarding the resolution for AMV I would keep it unchanged for SEVIRI and use the value stored in the BUFR file for FCI (but in the yaml-file). For the same reason above, there might be a reason the value is not set to `None`

Reply:
Resolution was added to the yaml files